podman/domain/containers.py

```diff
@@ -170,10 +170,24 @@ def exec_run(
             NotImplementedError: method not implemented.
             APIError: when service reports error
         """
-        if user is None:
-            user = "root"
-
-        raise NotImplementedError()
+        user = user or "root"
+        if isinstance(environment, dict):
+            environment = [f"{k}={v}" for k, v in environment.items()]
+        params = {
+            "AttachStderr": stderr,
+            "AttachStdin": stdin,
+            "AttachStdout": stdout,
+            "Cmd": cmd if isinstance(cmd, list) else [cmd],
+            # "DetachKeys": detach,
+            "Env": environment,
+            "Privileged": privileged,
+            "Tty": tty,
+            "User": user,
+            "WorkingDir": workdir
+        }
+        response = self.client.post(f"/containers/{self.name}/exec", params=params)
```

Comment:
@JacobCallahan Please review https://docs.podman.io/en/latest/_static/api.html#operation/ContainerExecLibpod, `name` is the only query parameter you may pass, all the other fields need to be passed as `data`.

Reply:
addressed in current update